core/templates/dev/head/domain/exploration/EditableExplorationBackendApiService.js

```diff
@@ -0,0 +1,161 @@
+  'ReadOnlyExplorationBackendApiService',
+  function($http, $q, EXPLORATION_DATA_URL_TEMPLATE,
+    EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, UrlInterpolationService,
+    ReadOnlyExplorationBackendApiService) {
+    var _fetchExploration = function(
+        explorationId, applyDraft, successCallback, errorCallback) {
+      var explorationDataUrl = UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        });
+
+      params = {};
+      if (applyDraft) {
+        params.apply_draft = applyDraft;
+      }
+
+      $http.get(explorationDataUrl, {
+        params: params
+      }).then(function(response) {
+        var exploration = angular.copy(response.data);
+        if (successCallback) {
+          successCallback(exploration);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _updateExploration = function(
+        explorationId, explorationVersion, commitMessage, changeList,
+        successCallback, errorCallback) {
+      var editableExplorationDataUrl = (
+        UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        })
+      );
+
+      var putData = {
+        version: explorationVersion,
+        commit_message: commitMessage,
+        change_list: changeList
+      };
+      $http.put(editableExplorationDataUrl, putData).then(function(response) {
+        // The returned data is an updated exploration dict.
+        var exploration = angular.copy(response.data);
+
+        // Delete from the ReadOnlyExplorationBackendApiService's cache
+        // As the two versions of the data (learner and editor) now differ
+        ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+          explorationId, exploration);
+
+        if (successCallback) {
+          successCallback(exploration);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _deleteExploration = function(
+        explorationId, role, successCallback, errorCallback) {
+      var editableExplorationDataUrl = (
+        UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        })
+      );
+
+      params = {};
+      if (role) {
+        params.role = JSON.stringify(role);
+      }
+
+      $http['delete'](editableExplorationDataUrl, {
+        params: params
+      }).then(function() {
+        // Delete item from the ReadOnlyExplorationBackendApiService's cache
+        ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+          explorationId);
+        if (successCallback) {
+          successCallback({});
```

Comment:
You don't need to pass anything to ``successCallback``, ie just ``successCallback();`` is fine.